### PR TITLE
v0.6: Release notes

### PR DESCRIPTION
**Pull request by valeriabarra**
I did an initial pass at adding the release notes page.

### CODE REVIEW

File: doc/sphinx/source/releasenotes.rst [hunk 1]
```diff
@@ -10,45 +10,69 @@ for each release of libCEED.
...
-Backends available in this release, are the same ones available in :ref:`v0.5`.
-
+libCEED v0.6 contains numerous new features and examples, as well as expanded
+documentation in `this new website <https://libceed.readthedocs.io>`_.
+
+New features
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+* New Python interface using `CFFI <https://cffi.readthedocs.io/>`_ provides a nearly
+  1-1 correspondence with the C interface, plus some convenience features.  For instance,
+  data stored in the :cpp:type:`CeedVector` structure are available without copy as
```
Comment (valeriabarra) [inline doc/sphinx/source/releasenotes.rst]: These `:cpp:type:` directives have not worked properly

Comment (valeriabarra) [reply doc/sphinx/source/releasenotes.rst]: Sorry, I meant the `:cpp:func:` ones

File: doc/sphinx/source/releasenotes.rst [hunk 2]
```diff
@@ -10,45 +10,69 @@ for each release of libCEED.
...
+  :py:class:`numpy.ndarray`.
+* Linear QFunctions can be assembled as block-diagonal matrices (per quadrature point,
+  :cpp:func:`CeedOperatorAssembleLinearQFunction`) or to evaluate the diagonal
```
Comment (valeriabarra) [inline doc/sphinx/source/releasenotes.rst]: These ones. See my comment above.

File: doc/sphinx/source/releasenotes.rst (hunk 1 above)
Comment (jedbrown) [reply doc/sphinx/source/releasenotes.rst]: Hmm, they link correctly in the docs I generated locally.

Comment (jedbrown) [reply doc/sphinx/source/releasenotes.rst]: I'm checking on RTD; will post a link shortly.

Comment (jedbrown) [reply doc/sphinx/source/releasenotes.rst]: They work in the PDF too.

Comment (valeriabarra) [reply doc/sphinx/source/releasenotes.rst]: They look like this in mine
Screenshot from 2020-03-29 23-00-58

File: doc/sphinx/source/releasenotes.rst (hunk 2 above)
Comment (valeriabarra) [reply doc/sphinx/source/releasenotes.rst]: Screenshot from 2020-03-29 23-00-58

File: doc/sphinx/source/releasenotes.rst [hunk 3]
```diff
@@ -10,45 +10,70 @@ for each release of libCEED.
...
+  (:cpp:func:`CeedOperatorAssembleLinearDiagonal`).  These operations are useful for
+  preconditioning ingredients and are used in the libCEED's multigrid examples.
+* The inverse of separable operators can be obtained using
+  :cpp:func`CeedOperatorCreateFDMElementInverse` and applied with
+  :cpp:func`CeedOperatorApply`.  This is a useful preconditioning ingredient, especially
+  for Laplacians and related operators.
+* New functions: :cpp:func:`CeedVectorNorm`, :cpp:func:`CeedOperatorApplyAdd`,
+  :cpp:func:`CeedQFunctionView`, :cpp:func:`CeedOperatorView`.
+* Make public accessors for various attributes to facilitate writing composable code.
+* New backend: ``/cpu/self/memcheck/serial``.
```
Comment (valeriabarra) [inline doc/sphinx/source/releasenotes.rst]: Is this not the same `/cpu/self/ref/memcheck` that was already in v0.5?

File: doc/sphinx/source/releasenotes.rst (hunk 1 above)
Comment (jedbrown) [reply doc/sphinx/source/releasenotes.rst]: https://libceed.readthedocs.io/en/v0.6changelog/releasenotes.html#v0-6-mar-29-2020

File: doc/sphinx/source/releasenotes.rst (hunk 2 above)
Comment (jedbrown) [reply doc/sphinx/source/releasenotes.rst]: So weird. Old/mismatched version of Sphinx, perhaps?

File: doc/sphinx/source/releasenotes.rst (hunk 3 above)
Comment (jedbrown) [reply doc/sphinx/source/releasenotes.rst]: #418 is new in v0.6.

**Comment by jedbrown** [comment]
@jeremylt I amended the commit to mention `CeedElemRestrictionApply`.

File: doc/sphinx/source/releasenotes.rst (hunk 2 above)
Comment (valeriabarra) [reply doc/sphinx/source/releasenotes.rst]: I have 2.4.4 as per our doc/sphinx/requiremenets.txt

File: doc/sphinx/source/releasenotes.rst (hunk 1 above)
Comment (valeriabarra) [reply doc/sphinx/source/releasenotes.rst]: I keep seeing them like that even in the readthedocs version you linked. Not sure why

File: doc/sphinx/source/releasenotes.rst (hunk 2 above)
Comment (valeriabarra) [reply doc/sphinx/source/releasenotes.rst]: It's the colon after func! It's missing

Comment (valeriabarra) [reply doc/sphinx/source/releasenotes.rst]: On line 27-28

Comment (jedbrown) [reply doc/sphinx/source/releasenotes.rst]: Thanks! Fixed.

File: doc/sphinx/source/releasenotes.rst [hunk 4]
```diff
@@ -10,45 +10,70 @@ for each release of libCEED.
...
+Notable additions include:
+
+* :file:`examples/petsc/bpsraw.c` (formerly ``bps.c``): transparent CUDA support.
+* :file:`examples/petsc/bps.c` (formerly ``bpsdmplex.c``): performance improvements and
+  transparent CUDA support.
+* :file:`examples/petsc/multigrid.c`: new p-multigrid solver with algebraic multigrid
+  coarse solve.
+* :ref:`example-petsc-bps-sphere` (:file:`examples/petsc/bpssphere.c`): generalizations
+  of all CEED BPs to the surface of the sphere; demonstrates geometric dimension
+  different from topological dimension.
```
Comment (valeriabarra) [inline doc/sphinx/source/releasenotes.rst]: Do we want to mention the `area` example? It is in the docs

File: doc/sphinx/source/releasenotes.rst (hunk 2 above)
Comment (jedbrown) [reply doc/sphinx/source/releasenotes.rst]: https://libceed.readthedocs.io/en/v0.6changelog/releasenotes.html#v0-6-mar-29-2020

File: doc/sphinx/source/releasenotes.rst (hunk 4 above)
Comment (jedbrown) [reply doc/sphinx/source/releasenotes.rst]: Didn't seem that significant considering that full bps on the sphere were available, but I'll add if you think it's notable.

Comment (valeriabarra) [reply doc/sphinx/source/releasenotes.rst]: I would say yes for completeness sake

File: doc/sphinx/source/releasenotes.rst [hunk 5]
```diff
@@ -10,45 +10,70 @@ for each release of libCEED.
...
-
+libCEED v0.6 contains numerous new features and examples, as well as expanded
+documentation in `this new website <https://libceed.readthedocs.io>`_.
+
+New features
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+* New Python interface using `CFFI <https://cffi.readthedocs.io/>`_ provides a nearly
+  1-1 correspondence with the C interface, plus some convenience features.  For instance,
+  data stored in the :cpp:type:`CeedVector` structure are available without copy as
+  :py:class:`numpy.ndarray`.
```
Comment (valeriabarra) [inline doc/sphinx/source/releasenotes.rst]: Do we want to mention the jupyter-notebooks at all, as sort of educational/outreach material?

Comment (jedbrown) [reply doc/sphinx/source/releasenotes.rst]: Perhaps put a Binder link on the part that introduces the Python interface?

**Pull request merged**